tests/api/test_remote_engine.py

```diff
@@ -41,21 +41,25 @@ def job(connection, monkeypatch):
     """Mocks a job which completes after a certain number of requests."""
     refresh_counter = 0
 
-    def refresh(self):
+    @property
+    def finished(self):
         nonlocal refresh_counter
         refresh_counter += 1
         if refresh_counter >= REQUESTS_BEFORE_COMPLETED:
-            self._status = JobStatus.COMPLETED
+            self._details = {"status": "complete"}
+            return True
+        return False
+
+    _details = {"status": "open"}
 
     job = xcc.Job(id_="123", connection=connection)
-    job._details = {"status": "open"}
 
-    result = Result(np.array([[1, 2], [3, 4]]), is_stateful=False)
+    result = {"samples": np.array([[1, 2], [3, 4]]), "foo": [np.array([5, 6])]}
 
     monkeypatch.setattr(xcc.Job, "submit", mock_return(job))
-    monkeypatch.setattr(sf.engine.Job, "refresh", refresh)
-    monkeypatch.setattr(sf.engine.Job, "result", result)
-    monkeypatch.setattr(sf.engine.Job, "meta", {"foo": "bar"})
+    monkeypatch.setattr(xcc.Job, "result", result)
+    monkeypatch.setattr(xcc.Job, "_details", _details)
```

Comment:
We may as well just use this instead:
```suggestion
    job._details = {"status": "open"}
```